smb/src/main/java/ch/cyberduck/core/smb/SMBListService.java

```diff
@@ -70,13 +70,13 @@ public AttributedList<Path> list(final Path directory, final ListProgressListene
                     type.add(Type.file);
                 }
 
-                result.add(new Path(directory, fileName, type).withAttributes(attributes));
+                result.add(new Path(directory, fileName, type, attributes));
 
             }
             return result;
         }
         catch(SMBApiException e) {
-            throw new SmbExceptionMappingService().map(e);
+            throw new SMBExceptionMappingService().map(e);
```

Comment:
Replace with parameters `"Listing directory {0} failed", e, directory`

```suggestion
            throw new SMBExceptionMappingService().map("Listing directory {0} failed", e, directory);
```